packages/data-provider/src/repository/submittedRepository.ts

```diff
@@ -64,11 +66,11 @@ const repository = (dependencies: BaseDependencies) => {
 	};
 
 	return {
-		delete: async (data: SubmittedData, comment: string, userName: string) => {
-			const deletedRecord = await db.delete(submittedData).where(eq(submittedData.id, data.id));
+		deleteBySystemId: async (systemId: string, userName: string) => {
+			const deletedRecord = await db.delete(submittedData).where(eq(submittedData.systemId, systemId)).returning();
 
-			if (features?.audit?.enabled) {
-				await auditDeleteSubmittedData(data, comment, userName);
+			if (features?.audit?.enabled && deletedRecord) {
+				await auditDeleteSubmittedData(deletedRecord, userName);
```

Comment:
Delete records that are marked for deletion under the Submission. Audit records if option is enabled